Shop_Unity_base/Assets/Scripts/Shop/Model/ShopModel.cs

```diff
@@ -1,161 +1,136 @@
         {
-            PopulateInventory(16); //currently, it has 16 items
+            PopulateInventory(24);
             gold = STARTING_GOLD;
-        }
-    
-        private void PopulateInventory(int itemCount)
-        {
-            for (int index = 0; index < itemCount; index++)
-            {
-                Item item = new Item("item", "item", 10); //item name, item icon, cost
-                itemList.Add(item);
-            }
-        }
-   
-        //returns the selected item
-        public Item GetSelectedItem()
-        {
-            if (selectedItemIndex >= 0 && selectedItemIndex < itemList.Count)
-            {
-                return itemList[selectedItemIndex];
-            }
-            else
-            {
-                return null;
-            }
-        }
 
-        //attempts to select the given item, fails silently
-        public void SelectItem(Item item)
-        {
-            if (item != null)
-            {
-                int index = itemList.IndexOf(item);
-                if (index >= 0)
-                {
-                    selectedItemIndex = index;
-                }
-            }
-        }
- 
-        //attempts to select the item, specified by 'index', fails silently
-        public void SelectItemByIndex(int index)
-        {
-            if (index >= 0 && index < itemList.Count)
-            {
-                selectedItemIndex = index;
-            }
+            OnBuyEvent += OnCustomerBuyItem;
+            OnSellEvent += OnCustomerSellItem;
         }
 
-        //returns the index of the current selected item
-        public int GetSelectedItemIndex()
-        {
-            return selectedItemIndex;
-        }
-    
-        //returns a list with all current items in the shop.
-        public List<Item> GetItems()
-        {
-            return new List<Item>(itemList); //returns a copy of the list, so the original is kept intact, 
-                                             //however this is shallow copy of the original list, so changes in 
-                                             //the original list will likely influence the copy, apply 
-                                             //creational patterns like prototype to fix this. 
-        }
-      
-        //returns the number of items
-        public int GetItemCount()
+        ~ShopModel()
         {
-            return itemList.Count;
+            OnBuyEvent -= OnCustomerBuyItem;
+            OnSellEvent -= OnCustomerSellItem;
         }
 
-        //tries to get an item, specified by index. returns null if unsuccessful
-        public Item GetItemByIndex(int index)
+        private void PopulateInventory(int itemCount)
         {
-            if (index >= 0 && index < itemList.Count)
-            {
-                return itemList[index];
-            }
-            else
+            inventory = new Inventory(itemCount);
+
+            // TODO: could change to IFactory factory = new ItemFactory<WeaponFactory>(); ?
```

Comment:
I am not sure about this one. I usually prefer the concrete types.

Reply:
Noted, will probably scrap idea cause of time, Don't want to tamper with what I (hopefully) don't need.